src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1539,6 +1621,26 @@ EnsureCoordinator(void)
 }
 
 
+/*
+ * InsertPlaceholderCoordinatorRecord inserts a placeholder record for the coordinator
+ * to be able to create distributed tables on a single node.
+ */
+void
+InsertPlaceholderCoordinatorRecord(void)
+{
+	NodeMetadata nodeMetadata = DefaultNodeMetadata();
+	nodeMetadata.groupId = 0;
+	nodeMetadata.shouldHaveShards = false;
```

Comment:
why don't we set this to true? 

Pros for setting it to true:
- Reflects the reality
- Simplify the code a bit

Cons:
- Even if users add new nodes, they'd still use the coordinator. But, because of auto colocation, this is already the case. I think we may consider having the `master_drain_node ` here as well, otherwise it'll be hard for users to scale-out the single-node Citus anyway. We are implicitly not using the new nodes if the coordinator is used as a single node.

Reply:
makes sense